appveyor.yml

```diff
@@ -0,0 +1,170 @@
+      EXTRA_TEST_ARGS: "--exclude fortran"
+
+    - PYTHON: "C:\\Python35-x64"
+      PYTHON_VERSION: "3.5.x"
+      PYTHON_ARCH: "64"
+      CMD_IN_ENV: "cmd /E:ON /V:ON /C .\\ci\\appveyor\\run-with-visual-studio.cmd"
+      EXTRA_TEST_ARGS: "--exclude fortran"
+
+init:
+  - cmd: "ECHO RDP CONNECTION INFORMATION (IN CASE OF HANGING BUILD)"
+  - ps: "\
+      iex (\
+        (new-object net.webclient).DownloadString(\
+          'https://raw.githubusercontent.com/\
+           appveyor/ci/master/scripts/enable-rdp.ps1'\
+        ))"
+
+install:
+  # If there is a newer build queued for the same PR, cancel this one.
+  # The AppVeyor 'rollout builds' option is supposed to serve the same
+  # purpose but it is problematic because it tends to cancel builds pushed
+  # directly to master instead of just PR builds (or the converse).
+  # credits: JuliaLang developers.
+  - ps: "\
+    if (\
+      $env:APPVEYOR_PULL_REQUEST_NUMBER -and \
+      $env:APPVEYOR_BUILD_NUMBER -ne (\
+        (\
+          Invoke-RestMethod \
+            ` https://ci.appveyor.com/api/projects/\
+                $env:APPVEYOR_ACCOUNT_NAME/\
+                $env:APPVEYOR_PROJECT_SLUG/\
+                history?recordsNumber=50\
+        ).builds | ` Where-Object pullRequestId -eq \
+                                  $env:APPVEYOR_PULL_REQUEST_NUMBER\
+      )[0].buildNumber\
+    ) { ` \
+      throw \"There are newer queued builds \
+              for this pull request, failing early.\" \
+    }"
+  - cmd: ECHO "Filesystem root:"
+  - ps: "ls \"C:/\""
+  - cmd: ECHO "Installed SDKs:"
+  - ps: "ls \"C:/Program Files/Microsoft SDKs/Windows\""
+
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside the powershell script as it would require to restart
+  # the parent CMD process).
+  - cmd: "SET PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%"
+
+  # Check that we have the expected version and architecture for Python
+  - cmd: "python --version"
+  - cmd: "\
+      python -c \"import struct; \
+                  print('    {}-bit'.format(struct.calcsize('P') * 8))\""
+
+  # Upgrade to the latest version of pip to avoid it displaying warnings
+  # about it being out of date.
+  - cmd: "pip install --disable-pip-version-check --user --upgrade pip"
+  - cmd: "pip install wheel"
+  - cmd: "cinst cmake"
+
+  # Install the build dependencies of the project. If some dependencies contain
+  # compiled extensions and are not provided as pre-built wheel packages,
+  # pip will build them from source using the MSVC compiler matching the
+  # target Python version and architecture
+  - cmd: "%CMD_IN_ENV% pip install -r requirements.txt"
+  - cmd: "%CMD_IN_ENV% pip install -r requirements-dev.txt"
+
+build_script:
+  # Build the compiled extension
+  - cmd: "%CMD_IN_ENV% python setup.py build"
+
+test_script:
+  # workaround for CMake not wanting sh.exe on PATH for MinGW
```

Comment:
I found this: https://cmake.org/Bug/view.php?id=7870
which led me to this: https://cmake.org/Wiki/CMake_MinGW_Compiler_Issues
And here's Appveyor's take: http://help.appveyor.com/discussions/problems/3193-cmake-building-for-mingw-issue-with-git-shexe

### Root Cause

Appveyor starts tests with an environment that includes GIT and everything it needs on your `PATH`.  This includes a copy of `sh.exe`.  Presumably, appveyor needs git to perform the project checkout, and they probably want to keep GIT on the `PATH` in case your project needs to run some other git commands.

The true root cause of the issue is MinGW Make insisting on running commands with `sh` if it's in your `PATH`, which will not work with Makefiles generated using the MinGW Makefile generator. The only way to force MinGW Make to run using `cmd.exe` is to explicitly remove any directory with an executable `sh` from the `PATH`.

### Fix in CMake?

Not wanting to meddle with a user's environment (let alone their `PATH`) seems like a good reason for a CMake maintainer to leave the issue as-is, especially considering that this is only a problem for Windows users who've deliberately added coreutils to their `PATH` (like in appveyor) and the simplest solution in this case is to just use a different generator.

### Fix in Appveyor?

On the other hand, keeping a functional git environment in a user's `PATH` from the outset seems like a more important feature to a member of the Appveyor support staff than catering to the specific needs of CMake users, or anyone who needs to use MinGW over Visual Studio, for that matter.

### What Do?

So, it's not obvious to me who we should be asking to free us from having to work around the issue, ourselves.